scripts/run_all.py

```diff
@@ -49,41 +125,91 @@ def run_build() -> None:
     assert returncode == 0, f"process exited with error code {returncode}"
 
 
+def run_web(examples: List[str], separate: bool) -> None:
+    if separate:
+        processes: List[Tuple[Any, Any]] = []
+        # start all examples in parallel
+        for path in examples:
+            # each example gets its own viewer
+            viewer = Viewer(web=True).start()
+            example = run_py_example(path, viewer_port=viewer.port, wait=False)
+            processes.append((viewer, example))
+        # wait for all processes to finish, and close the viewers
+        for process in processes:
+            viewer, example = process
+            example.wait()
+            viewer.close()
+    else:
+        with Viewer(close=True, web=True) as viewer:
+            for example in examples:
+                run_py_example(example, viewer_port=viewer.port)
+
+
+def run_separate_with(examples: List[str], close: bool, start_example: Callable[[str, int, bool], Any]) -> None:
+    processes: List[Tuple[Any, Any]] = []
+    # start all examples in parallel
+    for path in examples:
+        # each example gets its own viewer
+        viewer = Viewer().start()
+        example = start_example(path, viewer.port, False)
+        processes.append((viewer, example))
+    # wait for all examples to finish
+    for process in processes:
+        process[1].wait()
+    # close viewers if requested
+    if close:
+        for process in processes:
+            process[0].close()
+
+
+def run_save(examples: List[str], separate: bool, close: bool) -> None:
+    with Viewer(close=True) as viewer:  # ephemeral viewer that exists only while saving
+        for example in examples:
+            run_py_example(example, viewer_port=viewer.port, save="out.rrd")
+
+    if separate:
+        run_separate_with(examples, close, lambda path, viewer_port, wait: run_saved_example(path, wait))
+    else:
+        with Viewer(close) as viewer:
+            for example in examples:
+                run_saved_example(example)
+
+
+def run_native(examples: List[str], separate: bool, close: bool) -> None:
+    if separate:
+        run_separate_with(examples, close, run_py_example)
+    else:
+        with Viewer(close) as viewer:
+            for example in examples:
+                run_py_example(example, viewer_port=viewer.port)
+
+
 def main() -> None:
     parser = argparse.ArgumentParser(description="Runs all examples.")
-    parser.add_argument("--skip-build", action="store_true", help="Skip building the Python SDK")
-    parser.add_argument("--web", action="store_true", help="Run all examples in a web viewer.")
+    parser.add_argument("--skip-build", action="store_true", help="Skip building the Python SDK.")
+    parser.add_argument("--web", action="store_true", help="Run examples in a web viewer.")
     parser.add_argument(
         "--save",
```

Comment:
I'm wondering if this should be split up to two flags:
- `--save`, which runs all the examples and saves them
- `--load`, which runs the examples using the saved `out.rrd` files

Currently it does both and I think it defeats the purpose of saving the recordings. It works this way currently because I was just porting the `py-run-all-rrd` script, and that's how it worked.

Reply:
Yeah, I think that split would be great. `--save` is useful on its own to generate .rrd files, --load is useful on its own to quickly test without having to wait for Python. Make sure they both can be used at the same time though! (`run_all.py --save --load`)

Reply:
Done!